src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -319,106 +303,129 @@ public void AddRequestConditionHeaders(RequestConditionHeaders conditionHeaderFl
 
             _conditionHeaderFlag = conditionHeaderFlag;
 
+            string nameInRequest;
+            InputParameter? inputParameter;
+            Parameter parameter;
             switch (conditionHeaderFlag)
             {
                 case RequestConditionHeaders.IfMatch | RequestConditionHeaders.IfNoneMatch:
-                    _orderedParameters.Add(new ParameterChain(KnownParameters.MatchConditionsParameter, KnownParameters.MatchConditionsParameter, KnownParameters.MatchConditionsParameter));
-                    AddReference(KnownParameters.MatchConditionsParameter.Name, null, KnownParameters.MatchConditionsParameter, serializationFormat);
+                    inputParameter = null;
+                    parameter = KnownParameters.MatchConditionsParameter;
+                    nameInRequest = parameter.Name;
                     break;
-                case RequestConditionHeaders.IfMatch:
-                case RequestConditionHeaders.IfNoneMatch:
-                    AddParameter(requestConditionRequestParameter, typeof(ETag));
+                case RequestConditionHeaders.IfMatch or RequestConditionHeaders.IfNoneMatch:
+                    inputParameter = requestConditionRequestParameter;
+                    parameter = Parameter.FromInputParameter(inputParameter, new CSharpType(typeof(ETag), true), _typeFactory);
+                    nameInRequest = inputParameter.NameInRequest;
                     break;
                 default:
-                    _orderedParameters.Add(new ParameterChain(KnownParameters.RequestConditionsParameter, KnownParameters.RequestConditionsParameter, KnownParameters.RequestConditionsParameter));
-                    AddReference(KnownParameters.RequestConditionsParameter.Name, null, KnownParameters.RequestConditionsParameter, serializationFormat);
+                    inputParameter = null;
+                    parameter = KnownParameters.RequestConditionsParameter;
+                    nameInRequest = parameter.Name;
                     break;
             }
+
+            _orderedParameters.Add(new ParameterChain(parameter, parameter, parameter));
```

Comment:
Can we have another constructor of `ParameterChain(parameter)`? so that:
```suggestion
            _orderedParameters.Add(new ParameterChain(parameter));
```

Reply:
I would prefer it to be explicit. Constructor with less parameters often considered to be default one, while in our case it is not: models and generated enums can't be parameter types in protocol methods.